### PR TITLE
Release configuration

### LINKED ISSUE #46: Configuration of the release
Release of the application should be easy to configure (also in docker container)

### PR DESCRIPTION
**Pull request by rslota**
Closes #46 
This PR depends on #43 , so if you are reading this, please review #43 first :)

### CODE REVIEW

File: lib/fennec/application.ex [hunk 1]
```diff
@@ -1,22 +1,52 @@
...
+  @app :fennec
+  @ip_addr_config_keys [:ip, :relay_ip]
+
   use Application
 
   def start(_type, _args) do
+    loglevel = Confex.get(@app, :loglevel, :info)
+    Logger.configure(level: loglevel)
+
+    IO.inspect Fennec.Auth.get_secret()
```
Comment (arkgil) [inline lib/fennec/application.ex]: What is this `inspect` for?

File: lib/fennec/helper.ex [hunk 2]
```diff
@@ -21,4 +21,10 @@ defmodule Fennec.Helper do
   def inet_to_string(addr) do
     Kernel.to_string(:inet.ntoa(addr))
   end
+
+  @spec inet_to_string(String.t) :: Fennec.ip
```
Comment (arkgil) [inline lib/fennec/helper.ex]: Spec name does not match function name

File: mix.exs [hunk 3]
```diff
@@ -20,15 +20,17 @@ defmodule Fennec.Mixfile do
...
      mod: {Fennec.Application, []}]
   end
 
   defp elixirc_paths(:test), do: ["lib", "test/helper.ex"]
   defp elixirc_paths(_),     do: ["lib"]
 
   defp deps do
-    [{:ex_doc, "~> 0.14", runtime: false, only: :dev},
+    [{:confex, "~> 2.0.1"},
+     {:mix_docker, "~> 0.3.0"},
```
Comment (arkgil) [inline mix.exs]: `:mix_docker` probably needs `runtime: false` annotation.

File: Dockerfile.release [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    bash \
+    bash-completion \
+    curl \
+    dnsutils \
+    vim && \
+    apt-get clean
+
+ENV FENNEC_UDP_BIND_IP=0.0.0.0 FENNEC_UDP_PORT=3478 MIX_ENV=prod \
+    FENNEC_TCP_BIND_IP=0.0.0.0 FENNEC_TCP_PORT=3479 MIX_ENV=prod \
+    REPLACE_OS_VARS=true SHELL=/bin/bash
```
Comment (arkgil) [inline Dockerfile.release]: Is `REPLACE_OS_VARS` used anywhere? I don't understand (from the disillery docs) what is it used for except env vard substitution in `vm.args`. Is it also able to substitute variables in `sys.config`?

**Comment by arkgil** [comment]
I've left some minor comments. Apart from that, looks awesome!

**Comment by arkgil** [comment]
Oh, and it would be helpful to introduce a guide in a README, which explains how to build a standalone release and release in Docker.

**Comment by rslota** [comment]
> Oh, and it would be helpful to introduce a guide in a README, which explains how to build a standalone release and release in Docker. 

I plan on doing so with separate PR that will only update docs.

File: Dockerfile.release (hunk 4 above)
Comment (rslota) [inline Dockerfile.release]: To be honest - this is something that original docker file did (from `mix docker.customize`). I left it there to be "compatible" with other images build with this tool. We are not using this var, but some users could I guess.

File: mix.exs (hunk 3 above)
Comment (rslota) [inline mix.exs]: Thanks!

File: lib/fennec/application.ex (hunk 1 above)
Comment (rslota) [inline lib/fennec/application.ex]: Just something I forgot to remove, thanks!

File: Dockerfile.release (hunk 4 above)
Comment (arkgil) [inline Dockerfile.release]: Right. I think makes distillery replace some vars in sys.config and vm.args before starting the release.

**Comment by rslota** [comment]
@arkgil All review-related fixes are done.

**Pull request merged**